service/collector/rate_limit.go

```diff
@@ -0,0 +1,253 @@
+package collector
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/controller/v9/key"
+	"github.com/giantswarm/azure-operator/service/credential"
+)
+
+const (
+	remainingReadsHeaderName  = "x-ms-ratelimit-remaining-subscription-reads"
+	remainingWritesHeaderName = "x-ms-ratelimit-remaining-subscription-writes"
+	resourceGroupName         = "azure-operator-empty-rg-for-metrics"
+	metricsNamespace          = "azure_operator"
+	metricsSubsystem          = "rate_limit"
+)
+
+var (
+	ReadsDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(metricsNamespace, metricsSubsystem, "reads"),
+		"Remaining number of reads allowed.",
+		[]string{
+			"subscription",
+			"clientid",
+		},
+		nil,
+	)
+	WritesDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(metricsNamespace, metricsSubsystem, "writes"),
+		"Remaining number of writes allowed.",
+		[]string{
+			"subscription",
+			"clientid",
+		},
+		nil,
+	)
+	ReadsErrorCounter prometheus.Counter = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
Correct me if I'm wrong but I think they need to be registered somewhere. Check `prometheus.MustRegister`.

Reply:
I think you are right. But that means this counter is not working? https://github.com/giantswarm/azure-operator/blob/af47ec54be13a7b1b04b776262b0c67770992ab8/service/collector/usage.go#L91-L96

It's the one I based my counter on.

Reply:
<img width="356" alt="Screenshot 2019-10-31 at 13 31 01" src="https://user-images.githubusercontent.com/4608994/67951077-b33fe300-fbe2-11e9-90e1-ebd99361c792.png">

I don't see metric for that one. Maybe there are no errors but I guess it isn't registered.